### PR TITLE
Added Documentation for Vectorization of Bitmaps

### PR DESCRIPTION
**Pull request by harshagrwl**

### CODE REVIEW

**Comment by harshagrwl** [comment]
Closes #30

File: docs/common/vectorization.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Synfig Studio comes with a feature of Vectorization of Bitmaps. In this tutorial, we will learn what is Vectorization feature and how to use it. :)
+
+What is Vectorization of Bitmaps?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+There are two common ways to represent a two-dimensional image:
+
+    * Raster Graphics or Bitmap Graphics
+    * Vector Graphics
+
+In Bitmap Graphics images are stored as a 2-D grid of pixels. This has some problems for example if a bitmap image is selected it becomes blurry and pixelated.
```
Comment (ice0) [inline docs/common/vectorization.rst]: >2-D

2d

>is selected

i think you mean "scaled"?

File: docs/common/vectorization.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    .. image:: ../images/convert_to_vector.png
+
+Clicking on this "Convert to Vector" will open a dialog box with the following settings:
+    
+    .. image:: ../images/vector_dialog_box.png
+    
+
+    * **Threshold** - sets the value of the darkest pixels to be taken into account to detect lines to be converted to vector
+    * **Acurracy** - sets how much the vector stroke will follow the shape of the original drawing lines. High values create more precise strokes but makes them more complex.
```
Comment (ice0) [inline docs/common/vectorization.rst]: accuracy

File: docs/common/vectorization.rst (hunk 1 above)
Comment (harshagrwl) [reply docs/common/vectorization.rst]: > 2d
I have updated the changes
> i think you mean "scaled"?
Yes

File: docs/common/vectorization.rst (hunk 2 above)
Comment (harshagrwl) [reply docs/common/vectorization.rst]: I have updated the changes

File: docs/common/vectorization.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+Algorithm
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The Algorithm of Converting Vector Graphics from Bitmaps consists of the following steps:
+    * **Polygonization**: A Polygon is extracted from the image which gives a minimal representation of image contours.
+    * **Skeletonization**: The image is effectively reduced to an image of lines and the essential structure of the image is not altered.
+    * **Convert to strokes**: Conversion of the sequence into strokes. `Hermite Curves <https://www.inf.ed.ac.uk/teaching/courses/cg/d3/hermite.html>`_ are used for creation of strokes.
+
+The following image depicts the skeleton and contour representation of a Maple Leaf image:
+    .. image:: ../images/vector_maple.png
+
```
Comment (ankit-kumar-dwivedi) [inline docs/common/vectorization.rst]: this image seems irrelevant for user documentation/tutorial. Maybe we can add it in dev documenatation later on

**Comment by ankit-kumar-dwivedi** [comment]
@harshagrwl please see my comment

File: docs/common/vectorization.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    
+    .. image:: ../images/vector_dialog_box.png
+    
+
+    * **Threshold** - sets the value of the darkest pixels to be taken into account to detect lines to be converted to vector
+    * **Accuracy** - sets how much the vector stroke will follow the shape of the original drawing lines. High values create more precise strokes but makes them more complex.
+    * **Despeckling** - ignores during the conversion small areas generated by the image noise; the higher the value, the larger the areas ignored.
+    * **Max thickness** - sets the maximum vector stroke thickness; if this value is low very thick lines will be converted in two centerline strokes defining the line outline; if this value is high, they will be converted in a single centerline stroke.
+    
+
```
Comment (ankit-kumar-dwivedi) [inline docs/common/vectorization.rst]: add a note we currently only support outline vectorization

**Comment by harshagrwl** [comment]
@ankit-kumar-dwivedi I have removed the image as well as have added the Note.

**Comment by ankit-kumar-dwivedi** [comment]
@ice I think we can merge this one. Please have a look.

**Comment by morevnaproject** [comment]
Please note my comment here - https://github.com/synfig/synfig-docs-dev/issues/30#issuecomment-593110545

**Comment by morevnaproject** [comment]
@ice0 I think this PR contains mostly usage-centric information, so it's better to have it in User Docs. 

Apologies again for delay - haven't managed to set up a user docs repo. Will do that tomorrow!

**Comment by morevnaproject** [comment]
Hi guys! I have created a repo for user documentation and started to filling basic structure - https://github.com/synfig/synfig-docs

Please add vectorization docs here - https://github.com/synfig/synfig-docs/blob/master/source/artwork_import/vectorization.rst

**Comment by harshagrwl** [comment]
@morevnaproject I have added the vectorization docs in the user docs repository and made a pull request. You can review the request here: https://github.com/synfig/synfig-docs/pull/1

**Comment by morevnaproject** [comment]
Thank you! I am closing this one then. ^__^

**Pull request closed without merging**